### ISSUE TITLE
Multiple accounts

### ISSUE THREAD

**Issue by alandrumo** [BottlecapDave/HomeAssistant-OctopusEnergy#223]
I have two different octopus accounts and two meters in my house. How difficult would it be to add another account and meter to the same home assistant instance?

**Comment by BottlecapDave** [comment]
Hello. I've had a couple of requests, but tbh it's low on my list as it would require quite a major re-write. One of the biggest issues is the limitation on HA integration UIs. I'll keep this issue open so other people can upvote it.

**Comment by rwbarrett** [comment]
Would also like this as have 2 accounts

**Comment by ronhogwarts** [comment]
Same, two accounts for me

**Comment by ACALVETT** [comment]
Me to, useful as i would like to manage and automate my mums energy as well.

**Comment by BottlecapDave** [comment]
I've created an initial beta release to support this, which can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/multi-account-support-beta-1.

I don't have multiple accounts so very hard for me to test. Because of this and that less than 1% of installs are requesting this feature, the presence of the beta does not guarantee an eventual release.

Multiple accounts with target rate sensors and cost sensors that span multiple accounts should be tested.

**Comment by ACALVETT** [comment]
Amazing! I will setup and feedback

**Comment by rwbarrett** [comment]
HACS tells me I have the beta installed but when I add the integration it only asks me for one account number and sets up as before with only the devises for that single account

**Comment by BottlecapDave** [comment]
@rwbarrett You'll need to configure an initial account first. This flow will not change as 99% of users will not need to setup more than one account. I've updated my comment above with installation and instructions on how to add additional accounts.

**Comment by ACALVETT** [comment]
Working for me, i will test more. One minor thing to note is the api key is the same so don't need to add it again for multiple accounts.

**Comment by rwbarrett** [comment]
Thanks - all working now

**Comment by markcocker** [comment]
The beta is work great for me. I now have two accounts configured: both with gas + electric meters, and one also has export meter and four export rate meters. Amazing. Thank you!

**Comment by alandrumo** [comment]
Working for me too. 👌 ☺️
Thank you for adding the support for this!

**Comment by PhillyGilly** [comment]
Can I use this to compare two tariffs? I'm on the octopus variable E7 tariff but I'd like to compare this with Agile using traget Rate Sensors before switching over. Thanks

**Comment by BottlecapDave** [comment]
This will not support this functionality as you'll result in clashes in entity names (which include account id or meter ids). There is checks in place to ensure you can't setup the same account multiple times. You can use the override sensors to compare tariffs, but this will not effect target rate sensors.

**Comment by alfwro13** [comment]
I agree with @PhillyGilly that would be a great enhancement - i.e. add the same account 2nd time and use the tariff override for cost comparison. But I understand why that is currently not possible with the sensors are named.

**Comment by BottlecapDave** [comment]
There is a separate ticket around being able to override the main tariff of your account which would have a cascading effect to everything including target rate sensors. But this would obv replace all data while the tariff has been override (which could cause confusion when looking at past data). As mentioned, no plans to support this ability with multiple accounts support.

**Comment by BottlecapDave** [comment]
Speaking of multiple account support, has anyone had any issues with it?

**Comment by markcocker** [comment]
> Speaking of multiple account support, has anyone had any issues with it?

I have setup 3 accounts as below and 4 rate meters. Automated joining saving session automations for all 3. Getting electric + gas spins data for all 3.

- Electric meter, electric meter export, gas meter, "charger" for Intelligent Octopus
- Electric meter, gas meter, "charger" for Intelligent Octopus
- Electric meter, gas meter

**Comment by rwbarrett** [comment]
No it’s been a working really well for me - will this be wrapped into the main releases if there are no reported issues?Sent from my iPhoneOn 7 Feb 2024, at 14:59, David Kendall ***@***.***> wrote:﻿
Speaking of multiple account support, has anyone had any issues with it?

**Comment by BottlecapDave** [comment]
If there's no issues that are related to having multiple accounts, then yes the idea is to merge the feature into the main release.

**Comment by BottlecapDave** [comment]
New beta release which includes features/bugfixes from v10.0.4 and v10.1.0. This can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/multi-account-support-beta-2

**Comment by sandersr** [comment]
Will this integration work if I have two electricity meters on the same account? Recently I asked Octopus to add 2nd meter and rather than creating a new account, they've added it to my primary account. In their API I can see two meters:

https://api.octopus.energy/v1/electricity-meter-points/{METER_1_ID}/meters/{METER_1_SERIAL_NUMBER}/consumption/
and
https://api.octopus.energy/v1/electricity-meter-points/{METER_2_ID}/meters/{METER_2_SERIAL_NUMBER}/consumption/

**Comment by BottlecapDave** [comment]
@sandersr if both meters have a valid tariff associated with them, then both should be retrieved and populated.

**Comment by markcocker** [comment]
I have an automation to join the saving sessions for multiple accounts when the octopus_energy_new_octoplus_saving_session event is fired. The first account is joined as expected, but the second one fails with `Account ineligible to join Saving Sessions event.`. I can go to the Octopus site for those accounts an manually sign up to the savings session.

I wondered if this is due to timing - perhaps Octopus does not make all accounts eligible at the same time and so the 2nd account fails?

Should this integration fire the same event for each account, so perhaps my automation needs to handle just one account per event?

```yaml
alias: Octopus saving session - joiner
description: Automatically join Octopus Saving Session Events
trigger:
  - platform: event
    event_type: octopus_energy_new_octoplus_saving_session
condition: []
action:
  - service: octopus_energy.join_octoplus_saving_session_event
    data:
      event_code: "{{ trigger.event.data[\"event_code\"] }}"
    target:
      entity_id: event.octopus_energy_X_XXXXXXXX_octoplus_saving_session_events
    continue_on_error: true
  - service: octopus_energy.join_octoplus_saving_session_event
    data:
      event_code: "{{ trigger.event.data[\"event_code\"] }}"
    target:
      entity_id: event.octopus_energy_X_XXXXXXXX_octoplus_saving_session_events
    continue_on_error: true

etc.

  - service: persistent_notification.create
    data:
      title: Octopus Saving Sessions Updated
      message: >
        Joined Octopus Energy saving session. It starts on
        {{trigger.event.data["event_start"].day}}/{{trigger.event.data["event_start"].month
        }} at {{trigger.event.data["event_start"].strftime('%H:%M') }}.
mode: single
```
<img width="870" alt="Screenshot 2024-03-02 at 13 51 22" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/15351000/6180e9a5-a4e5-4ffe-aff1-200fe4650018">

**Comment by alandrumo** [comment]
I have two accounts but have already joined today's session on both. I can
test this if you want?
I'm very interested in auto join of the sessions.
I've already got a heap of automations to manage my power use over the
session but sometimes i miss the notification on my phone or am too late to
join.

Alan

**Comment by martindell** [comment]
Thanks @BottlecapDave.  Works a treat with two Octopus accounts 👍

**Comment by RaoulSargent** [comment]
PERFECT :-)

I have two accounts now working:
1) Electric Export, Electric Import, Gas
2) Electric Import, Gas

Very easy to setup.

I was previously running version 10.1.4 with the 1) account mentioned above.
I went into HACS, re-downloaded the Beta-2, restarted HA
Then added a new instance of the Octopus plugin in HA Settings / Devices & Services / Add Integration
This gave me the new menu option to add account, which I did with the account ID for the 2) account
Bingo, all working as expected.

I also use the blueprint called "Octopus Energy - Join Saving Session" with account 1)
So I setup another instance of the blueprint for account 2)
Bingo, that too appears to have created the automation correctly.

So all-in-all a very smooth setup experience.

Well done and Thank you :-)
